packages/angular/ssr/src/utils/ng.ts

```diff
@@ -28,13 +29,20 @@ import { renderApplication, renderModule } from '@angular/platform-server';
 export function renderAngular(
   html: string,
   bootstrap: Type<unknown> | (() => Promise<ApplicationRef>),
-  url: string,
+  url: URL,
```

Comment:
Consider: Call this `unnormalizedUrl` and then switch `normalizedUrl` -> `url`. That way the default "url" concept is the normalized version. This makes it harder for future changes to this function to accidentally use the unnormalized version.

Reply:
Refactored a bit as technically the input URL is normalized.